src/credentials/CHIPCert.cpp

```diff
@@ -187,11 +170,28 @@ CHIP_ERROR ChipCertificateSet::LoadCert(TLVReader & reader, BitFlags<CertDecodeF
         // Enter the certificate structure...
         ReturnErrorOnFailure(reader.EnterContainer(containerType));
 
-        // Initialize an ASN1Writer and convert the TBS (to-be-signed) portion of the certificate to ASN.1 DER
-        // encoding.  At the same time, parse various components within the certificate and set the corresponding
-        // fields in the CertificateData object.
-        writer.Init(mDecodeBuf, mDecodeBufSize);
-        ReturnErrorOnFailure(DecodeConvertTBSCert(reader, writer, cert));
+        // If requested to generate the TBSHash, verify that the decode buffer was initialized.
+        if (decodeFlags.Has(CertDecodeFlags::kGenerateTBSHash))
+        {
+            uint8_t decodeBuf[kMaxCHIPCertDecodeBufLength];
```

Comment:
This is a lot of stack to add. This amount of stack may be fine, but it should be called-out.

@jmartinez-silabs Would it be possible to test that this PR doesn't overrun on your smallest part, in usage?

Reply:
Another option would be to call:
    `decodeBuf = reinterpret_cast<uint8_t *>(chip::Platform::MemoryAlloc(kMaxCHIPCertDecodeBufLength));`
and then release.

Reply:
`ScopedMemoryBuffer` is what you want. It's an RAII allocator over MemoryAlloc.

Reply:
Updated to use `ScopedMemoryBuffer`